video_store.py

```diff
@@ -0,0 +1,127 @@
+import requests
+import datetime
+
+class VideoStore:
+
+    def __init__(self, url="https://retro-video-store-api.herokuapp.com", selected_customer=None, selected_video=None):
+        self.url = url
+        self.selected_customer = selected_customer
+        self.selected_video = selected_video
+    
+    # Option 1: add a video
+    def add_video(self,title,release_date,total_inventory):
+
+        query_params = {
+            "title": title,
+            "release_date": release_date,
+            "total_inventory": total_inventory
+        }
+        response = requests.post(self.url+"/videos",json=query_params)
+        return response.json()  
+    
+    # Option 2: edit a video 
+    def update_video(self,video_id,title,release_date,total_inventory):
+
+        if not title: 
+            title = self.selected_video["title"]
+        if not release_date:
+            release_date = self.selected_video["release_date"]
+        if not total_inventory:
+            total_inventory = self.selected_video["total_inventory"]   
+
+        query_params = {
+            "title": title,
+            "release_date": release_date,
+            "total_inventory": total_inventory,
+        }
+        response = requests.put(
+            self.url+f"/videos/{video_id}",
+            json=query_params
+            )
+    
+        self.selected_video = response.json()
+        return response.json()
```

Comment:
One opportunity for improvement for many people for this project is handling error codes from the API requests. In general, when writing code to call an API, it's appropriate to check whether the request was successful or it triggered some sort of error. This is where HTTP response codes come in handy! We can usually assume that a 2XX response code means there was no error and any other response code means there was some sort of error. It's very common to have an if...elif...else chain to handle different ranges of response codes.

This applies to all the other API requests in this project as well.